Run`make ut` error, too many open files

## Enhancement


## Problem1

make ut

```
building task finish, count=4783, takes=36.943982531s
run all tasks takes 2m38.904517081s
open temp cover file error: open /tmp/cov1281420580/executor.TestQueryTime: too many open files
make: *** [Makefile:131: gotest_in_verify_ci] Error 1
```

`ulimit -n`  result is 1024, but there are 4000+ temporary files...

## Problem2

After running `make ut`, too many *.test.bin file left, takes more than 5G disk size:

```
find . -name "*.bin" | xargs ls -l | awk '{ total += $5 }; END { print "total size: ",total }'
total size:  5943474416
```

--- a/Makefile
+++ b/Makefile
@@ -115,15 +115,19 @@ explaintest: server_check
 ddltest:
 	@cd cmd/ddltest && $(GO) test -o ../../bin/ddltest -c
 
+CLEAN_UT_BINARY := find . -name '*.test.bin'| xargs rm -rf
+
 ut: tools/bin/ut tools/bin/xprog failpoint-enable
 	tools/bin/ut $(X) || { $(FAILPOINT_DISABLE); exit 1; }
 	@$(FAILPOINT_DISABLE)
+	@$(CLEAN_UT_BINARY)
 
 gotest: failpoint-enable
 	@echo "Running in native mode."
 	@export log_level=info; export TZ='Asia/Shanghai'; \
 	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -timeout 20m -cover $(PACKAGES_TIDB_TESTS) -coverprofile=coverage.txt -check.p true > gotest.log || { $(FAILPOINT_DISABLE); cat 'gotest.log'; exit 1; }
 	@$(FAILPOINT_DISABLE)
+	@$(CLEAN_UT_BINARY)
 
 gotest_in_verify_ci: tools/bin/xprog tools/bin/ut failpoint-enable
 	@echo "Running gotest_in_verify_ci"
@@ -131,6 +135,7 @@ gotest_in_verify_ci: tools/bin/xprog tools/bin/ut failpoint-enable
 	@export TZ='Asia/Shanghai'; \
 	tools/bin/ut --junitfile "$(TEST_COVERAGE_DIR)/tidb-junit-report.xml" --coverprofile "$(TEST_COVERAGE_DIR)/tidb_cov.unit_test.out" || { $(FAILPOINT_DISABLE); exit 1; }
 	@$(FAILPOINT_DISABLE)
+	clean_ut_binary
 
 race: failpoint-enable
 	@export log_level=debug; \

--- a/Makefile
+++ b/Makefile
@@ -127,7 +127,6 @@ gotest: failpoint-enable
 	@export log_level=info; export TZ='Asia/Shanghai'; \
 	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -timeout 20m -cover $(PACKAGES_TIDB_TESTS) -coverprofile=coverage.txt -check.p true > gotest.log || { $(FAILPOINT_DISABLE); cat 'gotest.log'; exit 1; }
 	@$(FAILPOINT_DISABLE)
-	@$(CLEAN_UT_BINARY)
 
 gotest_in_verify_ci: tools/bin/xprog tools/bin/ut failpoint-enable
 	@echo "Running gotest_in_verify_ci"

--- a/Makefile
+++ b/Makefile
@@ -135,7 +135,7 @@ gotest_in_verify_ci: tools/bin/xprog tools/bin/ut failpoint-enable
 	@export TZ='Asia/Shanghai'; \
 	tools/bin/ut --junitfile "$(TEST_COVERAGE_DIR)/tidb-junit-report.xml" --coverprofile "$(TEST_COVERAGE_DIR)/tidb_cov.unit_test.out" || { $(FAILPOINT_DISABLE); exit 1; }
 	@$(FAILPOINT_DISABLE)
-	clean_ut_binary
+	@$(CLEAN_UT_BINARY)
 
 race: failpoint-enable
 	@export log_level=debug; \

--- a/Makefile
+++ b/Makefile
@@ -115,7 +115,7 @@ explaintest: server_check
 ddltest:
 	@cd cmd/ddltest && $(GO) test -o ../../bin/ddltest -c
 
-CLEAN_UT_BINARY := find . -name '*.test.bin'| xargs rm -rf
+CLEAN_UT_BINARY := find . -name '*.test.bin'| xargs rm
 
 ut: tools/bin/ut tools/bin/xprog failpoint-enable
 	tools/bin/ut $(X) || { $(FAILPOINT_DISABLE); exit 1; }

--- a/tools/check/ut.go
+++ b/tools/check/ut.go
@@ -401,20 +401,7 @@ func collectCoverProfileFile() {
 		if file.IsDir() {
 			continue
 		}
-
-		f, err := os.Open(path.Join(coverFileTempDir, file.Name()))
-		if err != nil {
-			fmt.Println("open temp cover file error:", err)
-			os.Exit(-1)
-		}
-		defer f.Close()
-
-		profs, err := cover.ParseProfilesFromReader(f)
-		if err != nil {
-			fmt.Println("parse cover profile file error:", err)
-			os.Exit(-1)
-		}
-		mergeProfile(result, profs)
+		collectOneCoverProfileFile(result, file)
 	}
 
 	w1 := bufio.NewWriter(w)
@@ -435,6 +422,22 @@ func collectCoverProfileFile() {
 			os.Exit(-1)
 		}
 	}
+}
+
+func collectOneCoverProfileFile(result map[string]*cover.Profile, file os.DirEntry) {
+	f, err := os.Open(path.Join(coverFileTempDir, file.Name()))
+	if err != nil {
+		fmt.Println("open temp cover file error:", err)
+		os.Exit(-1)
+	}
+	defer f.Close()
+
+	profs, err := cover.ParseProfilesFromReader(f)
+	if err != nil {
+		fmt.Println("parse cover profile file error:", err)
+		os.Exit(-1)
+	}
+	mergeProfile(result, profs)
 }
 
 func mergeProfile(m map[string]*cover.Profile, profs []*cover.Profile) {

